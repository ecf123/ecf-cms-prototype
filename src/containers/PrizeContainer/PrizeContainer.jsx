import "./PrizeContainer.scss";
import MoreOptions from "../../components/MoreOptions/MoreOptions";
import Card from "../../components/Card/Card";

const PrizeContainer = ({
  prizeData,
  optionsData,
  handleEdit,
  handleDelete,
}) => {

  return (
    <div className="prize-container">
      <div className="prize-container__middle-column">
        <Card
          title={prizeData.heading}
          img={prizeData.image}
          dateOrTime={prizeData.date}
          cardInfo={prizeData.overview}
          links={ prizeData.skillPoints}
          condition={prizeData.isLocked}
        />
      </div>
      <div className="prize-container__right-column">
        <MoreOptions
          title={optionsData.title}
          description={optionsData.description}
          handleDelete={handleDelete}
          handleEdit={handleEdit}
        />
      </div>
    </div>
  );
};
export default PrizeContainer;
