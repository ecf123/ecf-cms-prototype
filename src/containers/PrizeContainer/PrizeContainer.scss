.prize-container {
    width: 100%;
    display: flex;
    justify-content: center;
    gap: 3.1875rem;
}