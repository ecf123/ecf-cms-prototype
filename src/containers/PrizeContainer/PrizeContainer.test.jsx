import { customRender } from "../../utils/testUtils";
import PrizeContainer from "./PrizeContainer";

it("Should match the snapshot", () => {
    const { container } = customRender(
      <PrizeContainer articleData={"prizeData"} optionsData={"optionsData"} handleEdit={"handleEdit"} handleDelete={"handleDelete"} />
    );
    expect(container).toMatchSnapshot();
  });