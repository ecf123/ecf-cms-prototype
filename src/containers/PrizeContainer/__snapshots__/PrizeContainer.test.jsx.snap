// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
<div>
  <div
    class="prize-container"
  >
    <div
      class="prize-container__middle-column"
    >
      <div
        class="card"
      >
        <a
          href="/*"
        >
          <img
            class="card__image"
            src="https://firebasestorage.googleapis.com/v0/b/ecf-future-hub.appspot.com/o/market-place%2Famazon-logo.png?alt=media&token=0c9914b9-b020-421c-acaa-e87ff25be8a0"
          />
        </a>
        <p
          class="card__date"
        >
          25-10-2022
        </p>
        <a
          class="card__header"
          href="/*"
        >
          <h2
            class="card__title"
          >
            Earn a Â£5 Amazon Voucher
          </h2>
          <img
            alt="Arrow Icon"
            class="card__icon"
            src="arrow-upper-right-black.svg"
          />
        </a>
        <p
          class="card__info"
        >
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Hic placeat quis blanditiis. Debitis placeat sunt, sapiente, a iste repudiandae hic necessitatibus natus inventore maxime, perspiciatis sed distinctio praesentium blanditiis ipsa? Lorem ipsum dolor sit amet consectetur, adipisicing elit. Hic placeat quis blanditiis. Debitis placeat sunt, sapiente, a iste repudiandae hic necessitatibus natus inventore maxime, perspiciatis sed distinctio praesentium blanditiis ipsa? Lorem ipsum dolor sit amet consectetur, adipisicing elit. Hic placeat quis blanditiis. Debitis placeat sunt, sapiente, a iste repudiandae hic necessitatibus natus inventore maxime, perspiciatis sed distinctio praesentium blanditiis ipsa?
        </p>
        <ul
          class="categories-menu"
        >
          <li
            class="categories-menu__item--marketplace"
          >
            <img
              alt="Icon"
              class="categories-menu__item--icons"
              src="going-up-graph-arrow-grey.svg"
            />
            400
          </li>
          <li
            class="categories-menu__item--marketplace"
          >
            <img
              alt="Icon"
              class="categories-menu__item--icons"
              src="trophy-icon-grey.svg"
            />
            1
          </li>
        </ul>
      </div>
    </div>
    <div
      class="prize-container__right-column"
    >
      <div
        class="more-options"
      >
        <h3
          class="more-options__title"
        >
          more options for article
           Options
        </h3>
        <p
          class="more-options__description"
        />
        <div
          class="more-options__buttons"
        >
          <button
            class="button light-grey round-border medium"
          >
            <div
              class="medium-img black-svg"
            >
              <svg>
                pen-icon.svg
              </svg>
            </div>
            <p
              class="text medium-text light-grey"
            >
              Edit
            </p>
          </button>
          <button
            class="button red round-border medium"
          >
            <div
              class="medium-img white-svg"
            >
              <svg>
                bin-icon.svg
              </svg>
            </div>
            <p
              class="text medium-text red"
            >
              Delete
            </p>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
